common/app/com/commercetools/sunrise/framework/viewmodels/content/wishlist/WishlistItemViewModelFactory.java

```diff
@@ -1,37 +1,58 @@
 package com.commercetools.sunrise.framework.viewmodels.content.wishlist;
 
+import com.commercetools.sunrise.framework.reverserouters.productcatalog.product.ProductReverseRouter;
 import com.commercetools.sunrise.framework.viewmodels.SimpleViewModelFactory;
-import io.sphere.sdk.products.ProductVariant;
 import io.sphere.sdk.shoppinglists.LineItem;
+import play.mvc.Call;
+
+import javax.inject.Inject;
 
 /**
  * This view model factory creates {@link WishlistItemViewModel} instances for the product variants of
  * the wishlists line items {@link LineItem#getVariant()}.
  */
-public class WishlistItemViewModelFactory extends SimpleViewModelFactory<WishlistItemViewModel, ProductVariant> {
+public class WishlistItemViewModelFactory extends SimpleViewModelFactory<WishlistItemViewModel, LineItem> {
+
+
+    @Inject
+    private ProductReverseRouter productReverseRouter;
+
 
     @Override
-    public final WishlistItemViewModel create(final ProductVariant productVariant) {
-        return super.create(productVariant);
+    public final WishlistItemViewModel create(final LineItem lineItem) {
```

Comment:
👍 yep sooner or later well need additional infos from the LineItem object and there is no need to add it little by little every time.